src/axom/sina/core/Document.cpp

```diff
@@ -1152,5 +1105,186 @@ conduit::Node appendDocumentToHDF5(const std::string &hdf5FilePath,
 #endif
 }
 
+//-----------------------------------------------------------------------------
+// Internal helpers for auto-detection
+//-----------------------------------------------------------------------------
+
+namespace internal
+{
+
+Protocol detectOutputProtocol(const std::string& filepath)
+{
+  size_t dotPos = filepath.find_last_of('.');
```

Comment:
Minor: We have an `endsWith()` string utility that might be useful here (but no need to use it unless you feel it simplifies the code).

See:
* https://github.com/LLNL/axom/blob/1ca471117c2eca35191d3ff6eb9850759c906ae5/src/axom/core/utilities/StringUtilities.hpp#L28-L39
* https://github.com/LLNL/axom/blob/1ca471117c2eca35191d3ff6eb9850759c906ae5/src/examples/shaping_tutorial/lesson_02/improved_inlet_metadata.cpp#L265-L275